### PR TITLE
mpd-notification: init at 0.8.7

### PR DESCRIPTION
**Pull request by CaitlinDavitt**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by apfelkuchen6 — CHANGES_REQUESTED** [review]
Thanks for contributing!

Could you remove the merge-commit and squash the `mpd-notification`-commits such that you have just two commits 

 - maintainers: add CaitlinDavitt
 - mpd-notification: init at 0.8.7
 
and in this order?

File: pkgs/tools/audio/mpd-notification/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    mv mpd-notification $out/bin
+
+    mkdir -p $out/lib/systemd/user
+    cp systemd/mpd-notification.service $out/lib/systemd/user
+  '';
+
+   meta = with lib; {
+    description = "Notifications for mpd";
+    homepage = "https://github.com/eworm-de/mpd-notification";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (CaitlinDavitt) [reply pkgs/tools/audio/mpd-notification/default.nix]: Hey Apfel, I think I have done everything you requested. Thanks for reviewing my first PR

File: pkgs/tools/audio/mpd-notification/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   ];
+
+   installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv mpd-notification $out/bin
+
+    mkdir -p $out/lib/systemd/user
+    cp systemd/mpd-notification.service $out/lib/systemd/user
```
Comment (apfelkuchen6) [inline pkgs/tools/audio/mpd-notification/default.nix:45]: I just noticed that the systemd unit contains a hardcoded path that needs to be substituted.

Can you also rename the PR to `mpd-notification: init at 0.8.7`?

Other than that, this should be good to merge

Comment (CaitlinDavitt) [reply pkgs/tools/audio/mpd-notification/default.nix:45]: Hmm the original daemon seems a bit funky on nix. Might just have to remove it for now. 
```
# (C) 2011-2023 by Christian Hesse <mail@eworm.de>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

[Unit]
Description=MPD Notification
Requires=dbus.socket
PartOf=graphical-session.target
# Do not require any service here! We do rely on mpd OR network (for
# a remote mpd instance). So let the user care.
# We want to order after, though. This makes sure the resource is
# available on start and mpd-notification can cleanly disconnect on
# system shutdown.
After=mpd.service network.target network-online.target
# Order after notification daemons to make sure it is stopped before.
After=dunst.service xfce4-notifyd.service
ConditionUser=!@system

[Service]
Type=notify
Restart=on-failure
ExecStart=/usr/bin/mpd-notification

[Install]
WantedBy=default.target
```

Comment (apfelkuchen6) [reply pkgs/tools/audio/mpd-notification/default.nix:45]: What is the issue? the path in `ExecStart=` can easily be patched with something like 
```nix
postPatch = ''
   substituteInPlace systemd/mpd-notification.service --replace /usr $out
 '';
```
The "unrelated" units in `After=` should not cause any issues -- this just means that if systemd starts these units, they will be ordered before `mpd-notification`. This does not declare a dependency.

Comment (CaitlinDavitt) [reply pkgs/tools/audio/mpd-notification/default.nix:45]: Ah ok, thank you very much, should be good now

**Review by apfelkuchen6 — APPROVED** [review]
Looks great!

**Comment by CaitlinDavitt** [comment]
Thank you for helping out!

**Pull request merged**